byline.go

```diff
@@ -145,17 +145,15 @@ func (lr *Reader) MapStringErr(filterFn func(string) (string, error)) *Reader {
 	})
 }
 
-// Each - processing each line without changing the line
+// Each - processing each line
```

Comment:
Does that mean that it's not the same backing array being overwritten in each step of `Each`? Oh, or is it that the backing array contains the full content, and you're presenting a moving window over it with each callback, where each byte slice is pointing to a different piece of the same backing array (here, the string supplied to `strings.NewReader`)?